### DISCUSSION TITLE
Add new components

### DISCUSSION THREAD

**Discussion by duncanmarkc542** [ra3xdh/qucs_s#296] [Q&A] [answered]
Is it possible for me to add a new transistor (2SB772) to the library from the B772 Datasheet?  If so, how?  I am looking for the builder function for the component.

**Comment by ra3xdh** [comment]
You don't have to modify the library to add a BJT device. Use need to find a SPICE modelcard for this transistor. But the extraction of the modelcard from datasheet or measurements is not supported by Qucs-S. The attached screenshot illustrates how to use the SPICE modelcard with Qucs-S. JFET is shown here but, the principle of .MODEL for BJT is the same. 
![image](https://github.com/ra3xdh/qucs_s/assets/4920080/cfe046a7-bbdb-4739-a0ae-e05fe57dd79a)

**Comment by duncanmarkc542** [comment]
Thank you for responding.
This is my screenshot:
[image: image.png]
The datasheet is attached.
I am new to this application and I could not figure out how to do what you
said above.
If you look at the attachment, the values don't match those presented in
the edit components properties window:
[image: image.png]
I am at a loss of what to do.

Thank you for your concern.

On Tue, Jul 25, 2023 at 11:48 AM Vadim Kuznetsov ***@***.***>
wrote:

> You don't have to modify the library to add a BJT device. Use need to find
> a SPICE modelcard for this transistor. But the extraction of the modelcard
> from datasheet or measurements is not supported by Qucs-S. The attached
> screenshot illustrates how to use the SPICE modelcard with Qucs-S. JFET is
> shown here but, the principle of .MODEL for BJT is the same.
> [image: image]
> <https://user-images.githubusercontent.com/4920080/255957386-cfe046a7-bbdb-4739-a0ae-e05fe57dd79a.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ra3xdh/qucs_s/discussions/296#discussioncomment-6541970>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZYWXD3XJ5NQ5UVMWZSY4BTXR7TDVANCNFSM6AAAAAA2XHMGOM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Mark C. Duncan
***@***.***

**Comment by ra3xdh** [comment]
The images are missing in your message. Try to reupload the images.

**Reply by duncanmarkc542** [reply]
[2SB772.PDF](https://github.com/ra3xdh/qucs_s/files/12164521/2SB772.PDF)

![image](https://github.com/ra3xdh/qucs_s/assets/108096399/0f35765e-4b10-42b9-b666-c26270e796cc)
![image](https://github.com/ra3xdh/qucs_s/assets/108096399/0e4218ac-4f2c-4272-8a26-ac911f4b6602)

**Comment by duncanmarkc542** [comment]
I used the github this time


On Tue, Jul 25, 2023 at 2:42 PM Vadim Kuznetsov ***@***.***>
wrote:

> The images are missing in your message. Try to reupload the images.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ra3xdh/qucs_s/discussions/296#discussioncomment-6543723>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZYWXD2TUKATELRJ6AUOF5DXSAHQNANCNFSM6AAAAAA2XHMGOM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Mark C. Duncan
***@***.***

**Comment by ra3xdh** [comment, marked as answer]
You need to have plain text SPICE model to use with Qucs-S. The data provided by datasheet is not sufficient. No one circuit simulation software can extract the SPICE model from the datasheet. SPICE model should look like this (example for 2N3904). It seems the SPICE model is not freely available for 2SB772 BJT device. My advice is to debug circuit using some similar device that has SPICE model available. 

~~~
.model 2N3904   NPN(Is=6.734f Xti=3 Eg=1.11 Vaf=74.03 Bf=416.4 Ne=1.259
+               Ise=6.734f Ikf=66.78m Xtb=1.5 Br=.7371 Nc=2 Isc=0 Ikr=0 Rc=1
+               Cjc=3.638p Mjc=.3085 Vjc=.75 Fc=.5 Cje=4.493p Mje=.2593 Vje=.75
+               Tr=239.5n Tf=301.2p Itf=.4 Vtf=4 Xtf=2 Rb=10)
~~~
